.changeset/olive-papers-hide.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+fix(i18n): ensure the redirect pathname (Location response header) is non-empty
```

Comment:
```suggestion
Fixes a bug in i18n, where Astro caused an infinite loop when a locale that doesn't have an index, and Astro falls back to the index of the default locale.
```

Not sure if this suggestion works. @razonyang Would you mind rewording the changeset from the user's point of view?